app/views/users/show.html.slim

```diff
@@ -1,9 +1,13 @@
+- subscription = Subscription.new(followee: user, follower: current_user)
+
 .row
   .columns
     h2 = title("User")
     p = user.username
 
-    - if current_user.subscribed_to?(user)
+    - if !user_signed_in?
+      = link_to "Sign in to subscribe", new_user_session_path
+    - elsif current_user.subscribed_to?(user)
```

Comment:
```suggestion
    - if current_user && current_user.subscribed_to?(user)
```
I think we should add this callback to the subscriptions controller.
```ruby
before_action :authenticate_user!
```
In this case, we can leave only Subscribe and Unsubscribe buttons. If a visitor will click to the Subscribe button, he will be redirected to the sign-in form.